src/Tizen.NUI.Components/Controls/Extension/SwitchExtension.cs

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System.ComponentModel;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// The SwitchExtension class allows developers to access the Switch's components and extend their behavior in various states.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class SwitchExtension : ButtonExtension
+    {
+        /// <summary>
+        /// Called immediately after the Switch creates the track part.
+        /// </summary>
+        /// <param name="switchButton">The Switch instance that the extension currently applied to.</param>
+        /// <param name="track">The created Switch's track part.</param>
+        /// <param name="style">The initial style that will be appled to Switch's track part.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual void OnCreateTrack(Switch switchButton, ref ImageView track, ImageViewStyle style)
```

Comment:
Can you explain why "track" arg should be a ref?
ref must be used internally by an external reference.
But "track" seems to always be made in here.
Can't you use "out" to here?